### DISCUSSION TITLE
2023-02-06-alacritty

### DISCUSSION THREAD

**Discussion by giscus** [tw93/tw93.github.io#135] [General]
# 改良了下传说中最快的终端 - Tw93

Hi，我是 Tw93，是一个前端工程师，会一点 Swift / Rust / Python，喜欢折腾各种新技术。Technology最近发现了 Alacritty，一个基于 Rust 使用 OpenGL 加速的跨平台终端仿真器，只有 5M 的样子，传说中最快的终端，不过是真的丑，想着要不要改造一下，看能否当做我的默认终端，最终成品还不错。

https://tw93.fun/2023-01-25/alacritty.html

**Comment by lofiCafe** [comment]
很漂亮的配置，谢谢分享，另外对于Tab模式，其实使用tmux更不错

**Comment by kiyoakii** [comment]
谢谢分享！

**Reply by mrzzcn** [reply]
`fish-config` => `fish_config`

**Reply by tw93** [reply]
谢谢反馈 已经修改

**Reply by zhufenggood** [reply]
Wezterm也不错也是Rust写的，功能更新激进一些，UI渲染稍微慢点，支持Tab、字体Font Ligatures连体字。

**Comment by mitoop** [comment]
JetBrains Mono 字体我这怎么也加载不了 :笑哭

**Reply by tw93** [reply]
确定已安装字体
然后谷歌看看

**Comment by Dorixed** [comment]
它可以和Fig搭配使用吗？

**Reply by tw93** [reply]
可以 你试试

**Reply by Dorixed** [reply]
貌似有点bug, 兼容性不太好

**Comment by aliyuv** [comment]
谢谢分享~

**Comment by josangmi** [comment]
fish默认终端怎么设置都不行，反而item2都成了。是有啥关键地方没注意吗？路径也修改成brew的安装路径。

**Reply by josangmi** [reply]
我把 Alacritty重新装了一下然后把Yml文件去掉重新下载就好了

**Reply by tw93** [reply]
还有另外一种解法，直接在  Alacritty 配置文件里面加上，应该也行
```bash
shell:
  program: /usr/local/bin/fish
```

**Comment by josangmi** [comment]
安装fisher的时候，可以试试这个
curl -sL https://raw.githubusercontent.com/jorgebucaran/fisher/HEAD/functions/fisher.fish | source && fisher install jorgebucaran/fisher

**Reply by tw93** [reply]
谢谢提醒，之前那个地址由于墙可能不少同学有问题，直接用你这个会通顺些

**Comment by yzh990918** [comment]
想问下录GIF的软件是啥 好丝滑~

**Reply by tw93** [reply]
搜索下 ScreenFlow ，还不错的一个软件

**Comment by howie6879** [comment]
感谢作者的分享，我这边有一个问题，我无法输入中文符号类似 ，。、，一输入自动变成 ,./ 请问作者会有这个问题么？

期望回复，谢谢

**Reply by tw93** [reply]
试了试的确有你说的这个问题，可以先输入一个中文看看，然后再打标点符号，这个有点点像 “反向优化了”，因为命令行几乎不会去打英文符号，猜测是故意处理的

**Reply by howie6879** [reply]
我在终端里面用vim写代码，文档部分还是得中文，这标点符号有点坑了，找issue也没找到解决方案，不过这速度是真的快，很喜欢。

**Reply by shellRaining** [reply]
我也遇到了这个问题，使用Mac自带的输入法是可以正常输入全角标点的，切换到其他输入法就无法正常输入，现在写文档只能切回去

**Comment by Qiutum** [comment]
终端还是得选 kitty

**Reply by tw93** [reply]
kitty 有一点 印度的味道，其实做得很不错了，但是有一个点，窗口的初始启动位置居然不是居中的，而且不能配置，此外启动速度的强迫症，导致我放弃了他

**Comment by jaylinwang** [comment]
最近使用 Warp，结合了 GPT 的能力还不错，可以试试

**Comment by alvinooc** [comment]
谢谢分享！～

**Comment by F-TD5X** [comment]
Alacritty 真的很快，但多tab不重要，icon也不重要，就 font ligatures 没有可让喜欢用FiraCode的我太难受了（即使有个fork，但还是有点问题用不好）

**Comment by deke8908** [comment]
在ARM架构的Mac上，Homebrew总是安装到/opt/homebrew，并将其二进制文件放到/opt/homebrew/bin中。使用ARM架构Mac同学，文中的配置文件无法直接使用，需要修改路径

**Reply by tw93** [reply]
感谢反馈，辛苦给一下解法
PS 哈哈 我下个月应该也有M2了，到时候就方便了

**Comment by sleepm** [comment]
fish 语法不兼容 bash
好多文档给的命令都是 bash
粘进 fish 发现不兼容就很难受
zsh 语法兼容 bash
启动速度没感觉多慢
ohmyzsh 安装也挺快的

**Comment by diandianti** [comment]
zsh如果启动过慢，说明插件装多了。如果对于启动时间有要求，不建议使用oh-my-zsh。可以试试zinit，异步加载，速度很快

**Reply by sleepm** [reply]
zinit 现在是 zi 了 （以前存的书签）看着好像不错
https://github.com/zdharma/zi Turbo

**Reply by diandianti** [reply]
好家伙，我是多久没有更新了

**Comment by Oterea** [comment]
为什么我配置的快捷键不生效啊 比如像配置cmd+-> 移动到行尾。死活不生效
[keyboard]
bindings = [
    { key = "ArrowRight", mods = "Command" ,char = "\x1bOF", mode = "AppCursor"}
]
